lib/g10_certification_test_kit/bulk_data_authorization.rb

```diff
@@ -0,0 +1,229 @@
+
+      This test returns an access token.
+    DESCRIPTION
+
+    id :bulk_data_authorization
+
+    input :bulk_token_endpoint,
+          title: 'Backend Services Token Endpoint',
+          description: 'The OAuth 2.0 Token Endpoint used by the Backend Services specification
+                        to provide bearer tokens.',
+          default: 'https://inferno.healthit.gov/bulk-data-server/auth/token'
+    input :bulk_client_id,
+          title: 'Bulk Data Client ID',
+          description: 'Client ID provided at registration to the Inferno application.'
+    input :bulk_scope,
+          title: 'Bulk Data Scopes',
+          description: 'Bulk Data Scopes provided at registration to the Inferno application.',
+          default: 'system/*.read'
+    input :bulk_encryption_method,
+          title: 'Encryption Method',
+          description: 'The server is required to suport either ES384 or RS384 encryption methods
+                        for JWT signature verification. Select which method to use.',
+          type: 'radio',
+          default: 'ES384',
+          options: {
+            list_options: [
+              {
+                label: 'ES384',
+                value: 'ES384'
+              },
+              {
+                label: 'RS384',
+                value: 'RS384'
+              }
+            ]
+          }
+    output :bearer_token
+
+    http_client :token_endpoint do
+      url :bulk_token_endpoint
+    end
+
+    # TODO: Create after implementing TLS Tester Class.
+    test do
+      title 'Authorization service token endpoint secured by transport layer security'
+      description <<~DESCRIPTION
+        [§170.315(g)(10) Test Procedure](https://www.healthit.gov/test-method/standardized-api-patient-and-population-services) requires that
+        all exchanges described herein between a client and a server SHALL be secured using Transport Layer Security (TLS) Protocol Version 1.2 (RFC5246).
+      DESCRIPTION
+      # link 'http://hl7.org/fhir/uv/bulkdata/export/index.html#security-considerations'
+      run { pass }
+    end
+
+    test do
+      title 'Authorization request fails when client supplies invalid grant_type'
+      description <<~DESCRIPTION
+        The Backend Service Authorization specification defines the required fields for the
+        authorization request, made via HTTP POST to authorization token endpoint.
+        This includes the `grant_type` parameter, where the value must be `client_credentials`.
+
+        The OAuth 2.0 Authorization Framework describes the proper response for an
+        invalid request in the client credentials grant flow:
+
+        ```
+        If the request failed client authentication or is invalid, the authorization server returns an
+        error response as described in [Section 5.2](https://tools.ietf.org/html/rfc6749#section-5.2).
+        ```
+      DESCRIPTION
+      # link 'http://hl7.org/fhir/uv/bulkdata/authorization/index.html#protocol-details'
```

Comment:
Should this link be included directly in the description?

Reply:
Not a 100% sure, but will defer to anyone with a strong opinion. I think the description gives a solid narrative view of why/what's being tested and though the link might add some specificity, I don't think it's crucial to point users in that direction -- especially if we're already linking an overview of the authorization documentation at the beginning of the group. Again though, I'm okay with whatever.

Reply:
Inferno-Program had a `link` method for providing a reference to the spec. @Jammjammjamm is `link` getting carried over to core? If not it seems like this can either be deleted or incorporated into the description.